.github/dependabot.yml

```diff
@@ -4,4 +4,9 @@ updates:
   directory: "/"
   schedule:
     interval: weekly
-  open-pull-requests-limit: 10
\ No newline at end of file
+  open-pull-requests-limit: 10
+- package-ecosystem: cargo
+  directory: "/"
+  schedule: 
+    interval: weekly
+  open-pull-requests-limit: 10
```

Comment:
We might want to limit this to 1 instead because otherwise dependabot will open a lot of PRs at the same time, and then because of conflicts we will not be able to merge them anyway. Limiting this at 1 would allow us to progressively merge all the dependency updates without conflicts.

Reply:
Dependabot will automatically resolve conflicts whenever a PR is merged (and if that fails then you can always manually tell it to regenerate a PR). Since there's only a Cargo.toml here, all dependabot PRs should be `-1/+1` PRs, on disjoint lines, so there shouldn't be merge problems. 

I'd prefer to have dependabot open all the PRs at once to get an overview (especially because I'm not sure what delay there will be between merging one PR and dependabot opening the next one). Is dependabot already enabled on any other rust-vmm crate? We should strive for consistency here, too.

Reply:
I didn't know that dependabot now is able to fix conflicts, interesting. The problem remains though that you will need to re-approve on push force, so even if dependabot fixes the conflict, you still need to manually go to re-approve them one by one. AFAIK, the only other repository where dependabot is enabled is the vhost-devices one, but there approvals are not needed to merge the PRs, which is something that they did with the purpose of not having to re-approve on push forces. This is not something that I would be comfortable doing for other critical rust-vmm components though. While I also like consistency, in this context I don't think we can apply it.

We can leave it as is and reconsider in the future if you feel strongly about it. Let's see how this works.

Reply:
Yeah, I'd be in favor of observing how this plays out first. I'm hoping that dependabot is clever enough to realize that if a PR merges cleanly into main, it will not do the whole rebase-and-force-push, and instead recognize that the PR is mergable after clicking "update branch", which preserves approvals.

Reply:
IIRC, after one request is merged, if I let dependabot rebase the changes by itself (i.e. without clicking "update branch"), then it used to discard all the previous approvals. While clicking on "update branch" would create a merge commit and preserve the approvals.

I have gotten over it now, after I accidentally enabled "merge without approvals" somehow. I am using it only for bot requests which pass the tests. I simply go and rebase/merge the changes without waiting for anyone else's approvals, which are anyway not that important for bot requests I feel.

Reply:
Aw, that's unfortunate :(. Thanks for letting me know! 

I suppose there would be the option to disable the automatic rebasing altogether and rely on "update branch", and if something can't be "update branch"'d then manually trigger a rebase with `@dependabot rebase` (because in those cases, we cant get around manual intervention anyway)

Reply:
@roypat can you update it to not do push forces in this case?

Reply:
Alright, did this in a follow-up: https://github.com/rust-vmm/vm-allocator/pull/61